core/src/main/java/org/apache/carbondata/core/datamap/IndexDataMapProvider.java

```diff
@@ -99,25 +103,24 @@ private DataMapFactory createIndexDataMapFactory(DataMapSchema dataMapSchema)
     return dataMapFactory;
   }
 
-  private DataMapFactory getDataMapFactoryByShortName(String providerName)
+  public static DataMapFactory getDataMapFactoryByShortName(String providerName)
       throws MalformedDataMapCommandException {
+    DataMapRegistry.registerDataMap(DataMapClassProvider.LUCENE.getClassName(),
```

Comment:
What is the need of specifing the LUCENE here, these classes are supposed to be independent on any specific datamap.